### PR TITLE
[SelectionDAG][X86] Fix the assertion failure in Release build after #91747

### PR DESCRIPTION
**Pull request by KanRobert**
In #91747, we changed the SDNode from `X86ISD::SUB` (FROM) to `X86ISD::CCMP`
(TO) in the DAGCombine. The value type of `X86ISD::SUB` can be `i8, i32`
while the value type of `X86ISD::CCMP` is i32. This breaks the assumption
that the value type should match after the combine and triggers the
error

```
SelectionDAG.cpp:10942: void
llvm::SelectionDAG::transferDbgValues(llvm::SDValue, llvm::SDValue,
unsigned int, unsigned int, bool): Assertion `FromNode && ToNode &&
"Can't modify dbg values"' failed.
```

when running tests

llvm/test/CodeGen/X86/apx/ccmp.ll
llvm/test/CodeGen/X86/apx/ctest.ll

in Release build when LLVM_ENABLE_ASSERTIONS is on.

In this patch, we fix it by creating a merged value.

### CODE REVIEW

**Comment by KanRobert** [comment]
Thanks a lot for the suggestion @phoebewang !

File: llvm/lib/Target/X86/X86ISelLowering.cpp [hunk 1]
```diff
@@ -49253,8 +49253,10 @@ static SDValue combineX86SubCmpForFlags(SDNode *N, SDValue Flag,
     return SDValue();
 
   SDValue X = SetCC.getOperand(1);
-  // Replace API is called manually here b/c the number of results may change.
-  DAG.ReplaceAllUsesOfValueWith(Flag, X);
+  // sub has two results while X only have one. DAG combine assumes the value
+  // type matches.
+  if (N->getNumValues() > 1)
```
Comment (phoebewang) [inline llvm/lib/Target/X86/X86ISelLowering.cpp:49258]: Doesn't `N` always have 2 results?

Comment (KanRobert) [reply llvm/lib/Target/X86/X86ISelLowering.cpp:49258]: `N` can be a `X86ISD::CMP` or `X86ISD::SUB`. When `N` is `CMP`, it only has one result.

Comment (phoebewang) [reply llvm/lib/Target/X86/X86ISelLowering.cpp:49258]: How about `if (N->getOpcode() == X86ISD::SUB)` for clarity?

**Review by phoebewang — APPROVED** [review]
LGTM with one comment.

**Comment by KanRobert** [comment]
> How about if (N->getOpcode() == X86ISD::SUB) for clarity?
Good suggestion. Done.

File: llvm/lib/Target/X86/X86ISelLowering.cpp (hunk 1 above)
Comment (KanRobert) [reply llvm/lib/Target/X86/X86ISelLowering.cpp:49258]: Good suggestion, done

**Comment by KanRobert** [comment]
I will merge it soon to prevent this failure from causing trouble to other developers

**Pull request merged**